metapool/sample_sheet.py

```diff
@@ -388,6 +353,145 @@ def merge(self, sheets):
+            subset = list(
+                set(KLSampleSheet._KL_SAMPLE_SHEET_DATA_COLUMNS) & set(
+                    out.columns))
+            out = out[subset]
+
+        # append the new 'Well_description' column, now that alternates have
+        # been removed and non-essential columns have been dropped.
+        out['Well_description'] = well_description
+
+        for column in KLSampleSheet._KL_SAMPLE_SHEET_DATA_COLUMNS:
+            if column not in out.columns:
+                warnings.warn('The column %s in the sample sheet is empty' %
+                              column)
+                out[column] = ''
+
+        return out
+
+    def _add_data_to_sheet(self, table, sequencer, lanes, assay, strict=True):
+        table = self._remap_table(table, strict)
+        if assay != _AMPLICON:
+            table['index2'] = sequencer_i5_index(sequencer, table['index2'])
+
+            self.Bioinformatics['BarcodesAreRC'] = str(
+                sequencer in REVCOMP_SEQUENCERS)
+
+        for lane in lanes:
+            for sample in table.to_dict(orient='records'):
+                sample['Lane'] = lane
+                self.add_sample(sample_sheet.Sample(sample))
+
+
+class AmpliconSampleSheet(KLSampleSheet):
+    def __init__(self, path=None):
+        super().__init__(path=path)
+        # _KL_AMPLICON_REMAPPER
+        self.remapper = {
+            'sample sheet Sample_ID': 'Sample_ID',
+            'Sample': 'Sample_Name',
+            'Project Plate': 'Sample_Plate',
+            'Well': 'well_id_384',
+            'Name': 'I7_Index_ID',
+            'Golay Barcode': 'index',
+            'Project Name': 'Sample_Project',
+        }
+
+
+class MetagenomicSampleSheet(KLSampleSheet):
+    def __init__(self, path=None):
+        super().__init__(path=path)
+        # _KL_METAGENOMIC_REMAPPER
+        self.remapper = {
+            'sample sheet Sample_ID': 'Sample_ID',
+            'Sample': 'Sample_Name',
+            'Project Plate': 'Sample_Plate',
+            'Well': 'well_id_384',
+            'i7 name': 'I7_Index_ID',
+            'i7 sequence': 'index',
+            'i5 name': 'I5_Index_ID',
+            'i5 sequence': 'index2',
+            'Project Name': 'Sample_Project',
+            'synDNA pool number': 'syndna_pool_number'
+        }
+
+
+class MetatranscriptomicSampleSheet(KLSampleSheet):
+    def __init__(self, path=None):
+        super().__init__(path=path)
+        # _KL_METATRANSCRIPTOMIC_REMAPPER
+        self.remapper = {
+            'sample sheet Sample_ID': 'Sample_ID',
+            'Sample': 'Sample_Name',
+            'Project Plate': 'Sample_Plate',
+            'Well': 'well_id_384',
+            'i7 name': 'I7_Index_ID',
+            'i7 sequence': 'index',
+            'i5 name': 'I5_Index_ID',
+            'i5 sequence': 'index2',
+            'Project Name': 'Sample_Project',
+            'synDNA pool number': 'syndna_pool_number'
+        }
```

Comment:
What is different between the metagenomic and metatranscriptomic remappers?

Reply:
Nothing right now but as we brought up earlier, once the classes are in place we'll want to refactor the remapping.